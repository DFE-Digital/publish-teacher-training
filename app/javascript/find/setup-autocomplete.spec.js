/**
 * @jest-environment jsdom
 */

import setupAutocomplete from './setup-autocomplete'

describe('initACachedProvidersAutocomplete', () => {
  beforeEach(() => {
    document.body.innerHTML = `
         <div id="outer-container">
           <label for="query">Enter something</label>
           <select id="query">
             <option value>Select a option</option>
             <option value="A">A</option>
             <option value="B">B</option>
             <option value="C">C</option>
           </select>
         </div>
       `

    setupAutocomplete('query')
  })

  it('should instantiate an autocomplete', () => {
    expect(document.querySelector('#outer-container')).toMatchSnapshot()
  })
})
