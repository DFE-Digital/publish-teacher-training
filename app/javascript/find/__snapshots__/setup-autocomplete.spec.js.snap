// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setupAutocomplete should instantiate an autocomplete 1`] = `
<div
  id="outer-container"
>
  
           
  <label
    for="query"
  >
    Enter something
  </label>
  
           
  <div>
    <div
      class="autocomplete__wrapper"
    >
      <div
        style="border: 0px; height: 1px; margin-bottom: -1px; margin-right: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
      >
        <div
          aria-atomic="true"
          aria-live="polite"
          id="query__status--A"
          role="status"
        >
          
        </div>
        <div
          aria-atomic="true"
          aria-live="polite"
          id="query__status--B"
          role="status"
        >
          
        </div>
      </div>
      
      <input
        aria-autocomplete="list"
        aria-describedby="query__assistiveHint"
        aria-expanded="false"
        aria-owns="query__listbox"
        autocomplete="off"
        class="autocomplete__input autocomplete__input--default"
        id="query"
        name=""
        placeholder=""
        role="combobox"
        type="text"
      />
      
      <ul
        class="autocomplete__menu autocomplete__menu--inline autocomplete__menu--hidden"
        id="query__listbox"
        role="listbox"
      >
        
      </ul>
      <span
        id="query__assistiveHint"
        style="display: none;"
      >
        When autocomplete results are available use up and down arrows to review and enter to select.  Touch device users, explore by touch or with swipe gestures.
      </span>
    </div>
  </div>
  <select
    id="query-select"
    style="display: none;"
  >
    
             
    <option
      value=""
    />
    
             
    <option
      value="A"
    >
      A
    </option>
    
             
    <option
      value="B"
    >
      B
    </option>
    
             
    <option
      value="C"
    >
      C
    </option>
    
           
  </select>
  
         
</div>
`;
